---
title: "Searching MMRF-Compass Project database"
package: MMRFBiolinks
output: BiocStyle::html_document
vignette: >
  %\VignetteIndexEntry{2. Searching MMRF-Compass Project database}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(dpi = 300)
knitr::opts_chunk$set(cache = FALSE)

knitr::opts_chunk$set(fig.width = 6, fig.height = 6, width=8)
```




```{r message=FALSE, warning=FALSE, include=FALSE}
library(TCGAbiolinks)
library(MMRFBiolinks)
library(SummarizedExperiment)
library(dplyr)
library(DT)

```


# Overview of MMRFBiolinks functions

<div class="panel panel-info">
<div class="panel-heading"><b> MMRFBiolinks Vs TCGABiolinks</b></div>
<div class="panel-body">


<b>MMRFBiolinks</b> relies on the usage of <b>TCGABiolinks functions</b> among which <b>TCGAbiolinks::GDCquery </b> function for searching MMRF-COMMPASS data available at GDC Data Portal. [TCGAbiolinks package](https://bioconductor.org/packages/TCGAbiolinks/)
which should be consulted for more detailed information.
Left side of the following Table shows MMRFBiolinks functions while right side shows the incorporated TCGABiolink functions, if any. If the right entry is empty, it means that the MMRFBiolinks function on the right entry is not provided byTCGABiolinks (i.e. it is a new function, not covered by TCGABiolinks package). 

In MMRFBiolinks, the function name allows to indentify the data source: the functions whose
name contains 'GDC' deal with NCI-GDC dataset while
those whose name contains 'RG' deal with MMRF Research Gateway (MMRF-RG) dataset (e.g MMRFGDC\_GetIdentifierByTherapy deal with NCI-GDC data,
MMRFRG\_SurvivalKM deals with MMRF-RG data).


| MMRFBiolinks                         | TCGABiolinks           |
|--------------------------------------|------------------------|
| MMRFGDC\_QuerySummary                |                        |
| MMRFGDC\_ProjectSummary              |                        |
| MMRFGDC\_QueryClinic                 | GDCquery\_Clinic       |
| MMRFGDC\_prepare                     | GDC\_prepare           |    
| MMRFGDC\_GetTherapyByIdentifier      |                        |
| MMRFRG\_GetBor                       |                        |
| MMRFGDC\_QuerySamples                |                        |
| MMRFRG\_GetBorPlot                   |                        |
| MMRFRG\_TimeBorPlot                  |                        |
| MMRFRG\_TreatBorDurationPlot         |                        |
| MMRFRG\_SurvivalKM                   |                        |
| MMRFRG\_VariantCountPlot             |                        |
| MMRFRG\_GetIDSamplebyVariant         |                        |
  

</div>
</div>


<div class="panel panel-info">
<div class="panel-heading"><b>Different sources for MMRF-COMMPASS data</b></div>
<div class="panel-body">


MMRF-Compass data can be retrieved from two different sources:
    
<p align="justify"> - **GDC Data Portal**: MMRF-Commpass information about data available at GDC Data Portal can be retrieved through *MMRFGDC_QuerySummary* function. MMRF-Commpass data can be queried and downloaded using rispectively *TCGABiolinks::GDCquery* and *TCGABiolinks::GDCdownload* belonging to [TCGAbiolinks package](https://bioconductor.org/packages/TCGAbiolinks/). </p>

<p align="justify"> - **MMRF-CoMMpass Researcher Gateway**:  MMRF-CoMMpass data can be directly downloaded from MMRF-Commpass Researcher Gateway that has more information compared to GDC Data Portal data (e.g Best overall response). So the previous one is only a subset of this last. Once logged, the user can download data from MMRF-CoMMpass Researcher Gateway and import them as a dataframe into R environment for the further analysis.</p>



</div>
</div>


# Getting information and searching MMRF-COMMPASS data from GDC Data Portal

## MMRFqueryGDC\_Summary

<div class="panel panel-info">
<div class="panel-heading"><b>Data Search</b></div>
<div class="panel-body">
<p align="justify">
 
<b>MMRFqueryGDC\_Summary</b> allows to get information (including the number of cases) about query obteined from <i>GDCquery</i> function belonging to <i>TCGABiolinks</i> package.
</p>

</div>
</div>


```{r results = 'hide', message=FALSE, warning=FALSE, eval = F}

query.mm <- GDCquery(project = "MMRF-COMMPASS", 
                     data.category = "Transcriptome Profiling",
                     data.type = "Gene Expression Quantification",
                     workflow.type = "STAR - Counts")




```
```{r eval=FALSE, echo=TRUE}
# Download 
GDCdownload(query.mm)
```


```{r echo=TRUE, message=FALSE, warning=FALSE}

summary<-MMRFGDC_QuerySummary(query.mm)

# Only first 100 to make faster
datatable(summary, rownames = TRUE)

```













## TCGABiolinks::GDCquery arguments for MMRF-CoMMpass Project
The useful arguments for searching  MMRF-CoMMpass Project data are:


**Arguments** | **Description**
-----|-----
Data.category| A valid data category in the list below 
Data.type| A data type to filter the files to download
Workflow.type| GDC workflow type
Access| Filter by access type. Possible values: controlled
experimental\_strategy| Filter to experimental strategy
barcode| A list of barcodes to filter the files to download
sample.type| A sample type to filter the files to download
















The arguments options for filtering **MMRF-COMMPASS** data are listed below:


| Data.category                | Data.type                           | Workflow Type                      | Access               | experimental_strategy                                               
|----------------------------- | ----------------------------------- | -------------------------------    | -------------------- |------------------------ |
| Transcriptome Profiling      | Gene Expression Quantification      |  HTSeq - Counts                    | Open / Controlled    | RNA-Seq                 |                 
|                              |                                     |  HTSeq - FPKM-UQ                   | Open / Controlled    | RNA-Seq                 |                       
|                              |                                     |  HTSeq - FPKM                      | Open / Controlled    | RNA-Seq                 |                      
|                              |                                     |  STAR - Counts                     | Open / Controlled    | RNA-Seq                 | 
|                              | Splice Junction Quantification      |  STAR - Counts                     | Open / Controlled    | RNA-Seq                 |                      
| Simple Nucleotide Variation  | Raw Simple Somatic Mutation         |  MuSE                              | Controlled           | WXS                     |
|                              |                                     |  SomaticSniper                     | Controlled           | WXS                     |
|                              |                                     |  VarScan2                          | Controlled           | WXS                     |
|                              |                                     |  Pindel                            | Controlled           | WXS                     |
|                              |                                     |  MuTect2                           | Controlled           | WXS                     |
|                              | Annotated Somatic Mutation          |  MuSE Annotation                   | Controlled           | WXS                     |
|                              |                                     |  VarScan2 Annotation               | Controlled           | WXS                     |
|                              |                                     |  Pindel Annotation                 | Controlled           | WXS                     |
|                              |                                     |  MuTect2 Annotation                | Controlled           | WXS                     |
|                              |                                     |  SomaticSniper Annotation          | Controlled           | WXS                     |
| Sequencing Reads             | Aligned Reads                       |  BWA with Mark Duplicates and BQSR | Controlled           | WXS / RNA-Seq / WGS     |
|                              |                                     |  STAR 2-Pass Genome                | Controlled           | WXS / RNA-Seq / WGS     |
|                              |                                     |  STAR 2-Pass Transcriptome         | Controlled           | WXS / RNA-Seq / WGS     |


Below we provide further details about *Sample.type* an *Barcode* arguments: 


### Sample.type argument

The options for the  **`sample.type`** field in **MMRF-COMPASS Project** are:

**sample\_type.code**|**sample\_type.def**
-----|-----
TRBM|Recurrent Blood Derived Cancer - Bone Marrow
TBM|Primary Blood Derived Cancer - Bone Marrow
NB|Blood Derived Normal
TB|Primary Blood Derived Cancer - Peripheral Blood
TRB|Recurrent Blood Derived Cancer - Peripheral Blood

Example:

```{r message=FALSE, warning=FALSE}
#library(DT)
query.mm<-GDCquery(project = "MMRF-COMMPASS",
                   data.category = "Transcriptome Profiling",
                   data.type = "Gene Expression Quantification",
                   workflow.type="HTSeq - FPKM",
                   sample.type="Primary Blood Derived Cancer - Peripheral Blood")


```
```{r  echo=TRUE, message=FALSE, warning=FALSE}

getResults(query.mm, cols = c("cases.submitter_id","sample_type","cases")) %>% DT::datatable(options = list(scrollX = TRUE, keys = TRUE))

```

### Barcode argument 

Example:

```{r message=FALSE, warning=FALSE}
library(DT)
query.mm<-GDCquery(project = "MMRF-COMMPASS",
                   data.category = "Transcriptome Profiling",
                   data.type = "Gene Expression Quantification",
                   workflow.type="HTSeq - FPKM",
                   barcode = c("MMRF_2473","MMRF_2111",
                                "MMRF_2362","MMRF_1824",
                                "MMRF_1458","MRF_1361",
                                "MMRF_2203","MMRF_2762",
                                "MMRF_2680","MMRF_1797"))


```

```{r  echo=TRUE, message=FALSE, warning=FALSE}

getResults(query.mm, cols = c("cases.submitter_id","sample_type","cases")) %>% datatable(options = list(scrollX = TRUE, keys = TRUE))

```

# Getting MMRF-CoMMpass data from MMRF Research Gateway
<p align="justify">Once logged in to the [MMRF Research Gateway](https://research.themmrf.org/) Web Portal, you can download dataset you are interested in and import it as a dataframe into R environment for the further analysis.
Note:actually, files containing the data used by MMRFBiolinks functions are: 

**File name** | **Description**
-----|-----
MMRF_CoMMpass_IA14_STAND_ALONE_TRTRESP.csv|containing the data about the response to treatment
MMRF_CoMMpass_IA14a_All_Canonical_Variants.txt|containing the data about variants
MMRF_CoMMpass_IA14_PER_PATIENT.csv|containing the data about (eg. age, sex, date of death, date of the last follow up)



